### PR TITLE
Add more hub features to the table

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Ref: https://github.com/2i2c-org/meta/issues/628

Adds info about following features:
- dedicated cluster
- dedicated nodepool inside shared cluster
- custom html pages
- gh-scoped-creds

After this, I plan to dive into the terrform files and extract features from there.

### CODE REVIEW

File: docs/helper-programs/generate-hub-features-table.py [hunk 1]
```diff
@@ -91,20 +52,56 @@ def get_allusers_feature_status(hub_config, daskhub_type, binderhub_type):
...
     domain = f"[{hub['domain']}](https://{hub['domain']})"
     return {
         "domain": domain,
+        "dedicated cluster": False if hub_count else True,
+        "dedicated nodepool": dedicated_nodepool,
         "authenticator": authenticator,
-        "user id anonymisation": anonymization,
-        "admin access to all user's home dirs": allusers,
+        "user anonymisation": anonymization,
+        "allusers access": allusers,
```
[the remark this answers is not in the document]
Comment (sgibson91) [reply docs/helper-programs/generate-hub-features-table.py:99]: Maybe "user dir access" if length of string is a concern?

**Review by sgibson91 — APPROVED** [review]

File: docs/helper-programs/generate-hub-features-table.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GeorgianaElena) [reply docs/helper-programs/generate-hub-features-table.py]: @sgibson91, I've changed it to `admin access to allusers dirs`. I think `allusers` is a keyword that links nicely to the infrastructure docs at https://infrastructure.2i2c.org/topic/infrastructure/storage-layer/#hub-user-mount.

**Comment by GeorgianaElena** [comment]
Thanks @sgibson91!

File: docs/helper-programs/generate-hub-features-table.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgibson91) [reply docs/helper-programs/generate-hub-features-table.py]: That makes more sense to me!

**Pull request merged**